contrib/to-a11y-string/test/to-a11y-string-spec.js

```diff
@@ -0,0 +1,368 @@
+                `"left parenthesis, 1, plus, 3, right parenthesis"`,
+            );
+        });
+
+        test("simple brackets", () => {
+            const result = renderString("[1 + 3]");
+            expect(result).toMatchInlineSnapshot(
+                `"open bracket, 1, plus, 3, close bracket"`,
+            );
+        });
+
+        test("nested parens", () => {
+            const result = renderString("(a + (b + c))");
+            expect(result).toMatchInlineSnapshot(
+                `"left parenthesis, a, plus, left parenthesis, b, plus, c, right parenthesis, right parenthesis"`,
+            );
+        });
+
+        test("stretchy parens around fractions", () => {
+            const result = renderString("\\left(\\frac{1}{x}\\right)");
+            expect(result).toMatchInlineSnapshot(
+                `"left parenthesis, start fraction, 1, divided by, x, end fraction, right parenthesis"`,
+            );
+        });
+    });
+
+    describe("enclose", () => {
+        test("\\cancel", () => {
+            const result = renderString("\\cancel{a}");
+            expect(result).toMatchInlineSnapshot(
+                `"start cancel, a, end cancel"`,
+            );
+        });
+
+        test("\\fbox", () => {
+            const result = renderString("\\fbox{a}");
+            expect(result).toMatchInlineSnapshot(`"start box, a, end box"`);
+        });
+
+        test("\\sout", () => {
+            const result = renderString("\\sout{a}");
+            expect(result).toMatchInlineSnapshot(`"start strikeout, a, end strikeout"`);
+        });
+    });
+
+    describe("exponents", () => {
+        test("simple exponent", () => {
+            const result = renderString("e^x");
+            expect(result).toMatchInlineSnapshot(
+                `"e, start superscript, x, end superscript"`,
+            );
+        });
+
+        test("^{\\circ} => degrees", () => {
+            const result = renderString("90^{\\circ}");
+            expect(result).toMatchInlineSnapshot(`"90, degrees"`);
+        });
+
+        test("^{\\degree} => degrees", () => {
+            const result = renderString("90^{\\degree}");
+            expect(result).toMatchInlineSnapshot(`"90, degrees"`);
+        });
+
+        test("^{\\prime} => prime", () => {
+            const result = renderString("f^{\\prime}");
+            expect(result).toMatchInlineSnapshot(`"f, prime"`);
+        });
+
+        test("^2 => squared", () => {
+            const result = renderString("x^2");
+            expect(result).toMatchInlineSnapshot(`"x, squared"`);
+        });
+
+        test("^3 => cubed", () => {
+            const result = renderString("x^3");
+            expect(result).toMatchInlineSnapshot(`"x, cubed"`);
+        });
+
+        test("log_2", () => {
+            const result = renderString("\\log_2{x+1}");
+            expect(result).toMatchInlineSnapshot(
+                `"log, start subscript, 2, end subscript, x, plus, 1"`,
```

Comment:
Two ideas for what I'd consider "more natural":
1. Specific to log, I could see saying `start base ... end base` instead of `start subscript ... end subscript`.
2. In general, when a subscript, `\cancel`, etc. is given a "short" argument (maybe one character?), you could imagine using the noun instead of start noun ... end noun.  For example, `x_2` could be `x, subscript, 2` instead of `x, begin subscript, 2, end subscript`.

Just ideas -- feel free to ignore, or just add as TODOs.

Reply:
I should be able to do the `start base ... end base` as part of this PR.  I'll add a TODO for simplifying short subscripts.  I think having a single ticket that collects all of the TODOs is probably the way to go.

Reply:
I think `supsub`s for integrals and summations could be improved as well.